service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/CreateFederatedIdentityStep.java

```diff
@@ -0,0 +1,243 @@
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    var msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
+    var containerServiceManager =
+        crlService.getContainerServiceManager(azureCloudContext, azureConfig);
+
+    UUID landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(workspaceId));
+    var aksCluster =
+        landingZoneApiDispatch
+            .getSharedKubernetesCluster(bearerToken, landingZoneId)
+            .orElseThrow(() -> new RuntimeException("Shared Kubernetes cluster not found"));
+
+    var aksApi =
+        kubernetesClientProvider.createCoreApiClient(
+            containerServiceManager,
+            azureCloudContext.getAzureResourceGroupId(),
+            aksCluster);
+
+    var oidcIssuer =
+        Optional.ofNullable(
+                containerServiceManager
+                    .kubernetesClusters()
+                    .getByResourceGroup(
+                        azureCloudContext.getAzureResourceGroupId(), getResourceName(aksCluster))
+                    .innerModel()
+                    .oidcIssuerProfile()
+                    .issuerUrl())
+            .orElseThrow(() -> new RuntimeException("OIDC issuer not found"));
+    var uamiClientId =
+        msiManager
+            .identities()
+            .getByResourceGroup(azureCloudContext.getAzureResourceGroupId(), uamiName)
+            .clientId();
+
+    // the code above was lookup and setup, now we are ready to create the federated identity and
+    // k8s service account
+    createFederatedCredentials(msiManager, azureCloudContext, k8sNamespace, uamiName, oidcIssuer);
+    try {
+      createK8sServiceAccount(aksApi, k8sNamespace, uamiName, uamiClientId);
+    } catch (ApiException e) {
+      if (e.getCode() == HttpStatus.CONFLICT.value()) {
+        logger.info("K8s service account already exists");
+      } else {
+        logger.info("Failed to create k8s service account", e);
+        if (e.getCode() >= HttpStatus.INTERNAL_SERVER_ERROR.value()) {
+          return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+        } else {
+          return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+        }
+      }
+    }
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  private void createK8sServiceAccount(
+      CoreV1Api aksApi, String k8sNamespace, String uamiName, String uamiClientId)
+      throws ApiException {
+    var aksServiceAccount =
+        new V1ServiceAccount()
+            .metadata(
+                new V1ObjectMeta()
+                    .annotations(Map.of("azure.workload.identity/client-id", uamiClientId))
+                    .name(uamiName) // name of the service account is the same as the user-assigned
+                    .namespace(k8sNamespace));
+
+    aksApi.createNamespacedServiceAccount(k8sNamespace, aksServiceAccount, null, null, null, null);
```

Comment:
Does this create the namespace if it does not exist, or must the namespace exist beforehand?